Jetchat/app/src/main/java/com/example/compose/jetchat/NavActivity.kt

```diff
@@ -17,72 +17,98 @@
+import com.example.compose.jetchat.conversation.backPressHandler
+import com.example.compose.jetchat.theme.JetchatTheme
 
 /**
- * Main activity for the app. Shows a drawer and a toolbar rendered with traditional Views, for now.
+ * Main activity for the app.
  */
 class NavActivity : AppCompatActivity() {
 
-    private lateinit var appBarConfiguration: AppBarConfiguration
-    private lateinit var drawerLayout: DrawerLayout
+    // Used for navigation events between fragments.
+    private val viewModel: MainViewModel by viewModels()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        setContent {
+            Providers(BackPressedDispatcherAmbient provides this) {
+                val scaffoldState = rememberScaffoldState()
 
-        drawerLayout = findViewById(R.id.drawer_layout)
-        val navView: NavigationView = findViewById(R.id.nav_view)
-        val navController = findNavController(R.id.nav_host_fragment)
-        // Passing each menu ID as a set of Ids because each
-        // menu should be considered as top level destinations.
-        appBarConfiguration = AppBarConfiguration(
-            setOf(
-                R.id.nav_home,
-                R.id.nav_profile
-            ),
-            drawerLayout
-        )
-        navView.setupWithNavController(navController)
-    }
+                val openDrawerEvent = viewModel.drawerShouldBeOpened.observeAsState()
+                if (openDrawerEvent.value == true) {
+                    // Open drawer and reset state in VM.
+                    scaffoldState.drawerState.open {
+                        viewModel.resetOpenDrawer()
+                    }
+                }
 
-    override fun onCreateOptionsMenu(menu: Menu): Boolean {
-        // Inflate the menu; this adds items to the action bar if it is present.
-        menuInflater.inflate(R.menu.main, menu)
-        return true
-    }
+                // Intercepts back navigation when the drawer is open
+                backPressHandler(
+                    enabled = scaffoldState.drawerState.isOpen,
+                    onBackPressed = { scaffoldState.drawerState.close() },
+                    highPriority = true
+                )
 
-    override fun onSupportNavigateUp(): Boolean {
-        val navController = findNavController(R.id.nav_host_fragment)
-        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
+                JetchatScaffold(scaffoldState)
+            }
+        }
     }
 
-    /**
-     * Back closes drawer if open.
-     */
-    override fun onBackPressed() {
-        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
-            drawerLayout.closeDrawer(GravityCompat.START)
-        } else {
-            super.onBackPressed()
+    @Composable
+    private fun JetchatScaffold(
+        scaffoldState: ScaffoldState,
```

Comment:
Nit: Give a default value

```suggestion
        scaffoldState: ScaffoldState = rememberScaffoldState(),
```